### ISSUE TITLE
C++ unit test for boost ICU regex support

### ISSUE THREAD

**Issue by springmeyer** [mapnik/mapnik#3483]
@artemp We need a unit test in C++ that will fail when Mapnik is using boost regex without icu support. I'm noticing we are missing this because currently python-mapnik has a single failing test due to being built against a mapnik that is built against a boost_regex lib not linked to ICU. 

```
FAIL: python_tests.filter_test.test_unicode_regex_replace
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/mapnik/python-mapnik/mason_packages/.link/lib/python2.7/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/home/travis/build/mapnik/python-mapnik/test/python_tests/filter_test.py", line 193, in test_unicode_regex_replace
    eq_(expr.evaluate(f), u'Q u é b e c')
AssertionError: u'Q u\ufffd \ufffdb e c' != u'Q u \xe9 b e c'
```
https://travis-ci.org/mapnik/python-mapnik/builds/150714641#L2816

This is expected because the mason package for boost lacks this (which needs to be fixed). So lets:

 - [x] Add a C++ unit test that should fail in mapnik core (in a branch) @artemp 
 - [x] Then fix the mason package to provide icu support @springmeyer
 - [x] Then merge that branch into master once the branch is green.

**Comment by artemp** [comment]
@springmeyer - more tests are in `regex-unicode` branch but I'd like to sync up per logic ^

**Comment by springmeyer** [comment]
@artemp thanks. Can you say more about what you mean by 'sync up per logic'?

**Comment by artemp** [comment]
@springmeyer - oops sorry wasn't clear ^ try again => why not just check for `BOOST_REGEX_HAS_ICU` and put `#ifdef/#endif` around tests that are only expected to work with ICU support ? Building mapnik without ICU is a valid configuration and failing tests doesn't sounds right to me.

**Comment by springmeyer** [comment]
> failing tests doesn't sound right to me.

I see your point, but disagree. This is such a critical feature for rendering I think a failing test is worth it.

**Comment by springmeyer** [comment]
Merged: #3485
